docs/types.rst

```diff
@@ -337,7 +337,8 @@ function type should not return anything, the whole ``returns (<return types>)``
 part has to be omitted.
 
 By default, function types are internal, so the ``internal`` keyword can be
-omitted.
+omitted. Note that this is not the case for functions defined on a contract,
```

Comment:
Hmmm... I see what you mean. Confusion seems to root from the overlap between public/private function types, and public/private/internal/external functions. I agree that explaining the differences first, and providing some examples of conversions between the two, would make this clarification later on a lot clearer. As-is it can cause more confusion, which is obviously not what I intended to do here.